gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,1441 @@
+        "length": u.dimensions.length,
+        "mass": u.dimensions.mass,
+    }
+
+    unit_systems = {"MD": MD_UNITS, "AKMA": AKMA_UNITS}
+    if base_units and auto_scale:
+        warnings.warn(
+            "Both base_units and auto_scale are provided, auto_scale will take precedent."
+        )
+
+    if auto_scale:
+        base_units = _infer_units(top)
+
+        # Refer masses from sites' masses
+        masses = [site.mass.to(base_units["mass"]) for site in top.sites]
+        if masses:
+            base_units["mass"] = max(masses)
+            # Refer lengths and energies from sites' atom types if possible
+        unique_atypes = top.atom_types(
+            filter_by=PotentialFilters.UNIQUE_NAME_CLASS,
+        )
+        if unique_atypes:
+            if not potential_types:
+                potential_types = _validate_compatibility(top)
+            atype_classes = dict()
+            # Separate atypes by their classes
+            for atype in unique_atypes:
+                if potential_types[atype] not in atype_classes:
+                    atype_classes[potential_types[atype]] = [atype]
+                else:
+                    atype_classes[potential_types[atype]].append(atype)
+
+        # Appending lenghts and energy
+        lengths, energies = list(), list()
+        for atype_class in atype_classes:
+            if atype_class == "LennardJonesPotential":
+                for atype in unique_atypes:
+                    lengths.append(
+                        atype.parameters["sigma"].to(base_units["length"])
+                    )
+                    energies.append(
+                        atype.parameters["epsilon"].to(base_units["energy"])
+                    )
+            else:
+                raise NotYetImplementedWarning(
+                    f"Currently cannot infer referenced lengths and energies from {atype_class}"
+                )
+        base_units["length"] = max(lengths)
+        base_units["energy"] = max(energies)
+
+    elif isinstance(base_units, str):
+        base_units = unit_systems[base_units]
+    elif isinstance(base_units, dict):
+        for key in base_units:
+            if key not in ["energy", "mass", "length"]:
+                warnings.warn(
+                    "Only base unit will be used during the conversion "
+                    "i.e., energy, mass, and length, other units provided "
+                    "will not be considered."
+                )
+            else:
+                msg = "{key} is in wrong unit dimension"
+                if isinstance(base_units[key], u.Unit):
+                    assert base_units[key].dimensions == ref[key], msg
+                elif isinstance(base_units[key], u.Unit):
+                    assert base_units[key].units.dimensions == ref[key], msg
+                else:
+                    raise TypeError(
+                        f"Base unit of {key} must be of type u.Unit or u.unyt_quantity."
```

Comment:
@daico007 Quick quesiton, if I try using `unyt.amu` for the mass base units, this exception is being raised.

I'm not very familiar with Unyt, but for some reason `unyt.amu` has a different type than `unyt.g` (is there a reason for this that you know of?)

`unyt.amu` is of type `unyt.array.unyt_quantity`. For comparison, `unyt.g`  and `unyt.g/unyt.mol` are type `unyt.unit_object.Unit`

Do we want to allow someone to use a mass unit of `amu` here. I know it's the same as as `g/mol`, which works, but some people might default to using `amu`; then get confused about this error message.

Reply:
Good point! I didn't consider the amu when doing the parser. I will at a special treatment for `u.amu`.

Reply:
I think I found the bug for this, working on a fix now.